domain/application/state/unit.go

```diff
@@ -916,16 +916,23 @@ func (st *State) insertUnit(
 		return "", "", errors.Capture(err)
 	}
 	createParams := unitDetails{
-		ApplicationID: appUUID,
-		UnitUUID:      unitUUID,
-		Name:          args.UnitName,
-		NetNodeID:     nodeUUID.String(),
-		LifeID:        life.Alive,
+		ApplicationID:  appUUID,
+		UnitUUID:       unitUUID,
+		Name:           args.UnitName,
+		NetNodeID:      nodeUUID.String(),
+		LifeID:         life.Alive,
+		OSID:           int(args.Platform.OSType),
+		ArchitectureID: int(args.Platform.Architecture),
```

Comment:
question: What about the Platform.Channel? Will that be saved?

Reply:
The issue I have with storing the platform's channel there is that it will be confusing. Units being different than applications because apps can have different bases than the current units do (as you explained in https://github.com/juju/juju/pull/19380#issuecomment-2770649551), but a single unit should never have two different bases. So I opted for not storing it.